server/src/infra/infra.utils.ts

```diff
@@ -91,3 +121,79 @@ export function ChunkedArray(options?: { paramIndex?: number }): MethodDecorator
 export function ChunkedSet(options?: { paramIndex?: number }): MethodDecorator {
   return Chunked({ ...options, mergeFn: setUnion });
 }
+
+export function searchAssetBuilder(
+  builder: SelectQueryBuilder<AssetEntity>,
+  options: AssetSearchBuilderOptions,
+): SelectQueryBuilder<AssetEntity> {
+  builder.andWhere(
+    _.omitBy(
+      {
+        createdAt: OptionalBetween(options.createdAfter, options.createdBefore),
+        updatedAt: OptionalBetween(options.updatedAfter, options.updatedBefore),
+        deletedAt: OptionalBetween(options.trashedAfter, options.trashedBefore),
+        fileCreatedAt: OptionalBetween(options.takenAfter, options.takenBefore),
+      },
+      _.isUndefined,
+    ),
+  );
+
+  const exifInfo = _.omitBy(_.pick(options, ['city', 'country', 'lensModel', 'make', 'model', 'state']), _.isUndefined);
+  if (Object.keys(exifInfo).length > 0) {
+    builder.leftJoin(`${builder.alias}.exifInfo`, 'exifInfo');
+    builder.andWhere({ exifInfo });
+  }
+
+  const id = _.pick(options, ['checksum', 'deviceAssetId', 'deviceId', 'id', 'libraryId', 'ownerId']);
+  builder.andWhere(_.omitBy(id, _.isUndefined));
+
+  const path = _.pick(options, ['encodedVideoPath', 'originalFileName', 'originalPath', 'resizePath', 'webpPath']);
```

Comment:
I guess we could technically type those array, e.g. `Array<keyof SearchPathOptions>`. Not sure if it's worth it though

Reply:
I typed them like that earlier. The only reason I removed that is to keep them on one line haha.